src/action-sheets/index.js

```diff
@@ -449,6 +456,9 @@ export const constructMessageActionButtons = ({
     && (isStreamOrTopicNarrow(narrow) || isPmNarrow(narrow))
   ) {
     buttons.push(editMessage);
+    moveMessage.narrow = narrow;
+    moveMessage.admin = ownUser.is_admin;
```

Comment:
This appears to be setting properties on the function (which is a global constant for this module), in order to store some of the data that we're currently processing.

Do not do this. This is a brittle pattern -- effectively you're storing these values as global variables.

Instead, use the mechanisms that the surrounding code already uses to pass the data that comparable functions need. Here, `moveMessage` needs to be a `Button<MessageArgs>`, because that's what's in the array that `constructMessageActionButtons` returns. Looking at the definition of `Button<Args>`, in general the function takes an `Args` -- so for `Button<MessageArgs>` it takes a `MessageArgs`. So we just need the relevant information to be added to `MessageArgs`.

Specifically we need `narrow` there. For `ownUser.is_admin`, the `moveMessage` callback doesn't really need that information -- the `MoveMessage` component will need it, but it can get that information from Redux for itself.

Reply:
BTW in general if you see a way to make the code work that involves structuring it with a bad pattern, and you can't find a cleaner way to do it, it's OK to send a PR where it uses the bad structure (especially when it's a small part of the PR's code, like here) -- *but* it's important to call attention to it clearly. Typically that might mean
* both a TODO comment in the code,
* and a prominent note about it in the PR description, or as a PR comment shortly after sending the PR.

That way you help the reviewer make sure to notice it so they can offer suggestions on how to handle the problem.

Better yet, typically, is to ask in chat about how to handle the thing you're trying to do, before you have the PR ready to send. But sometimes the easiest way to make the question clear is with the context of the rest of the code, and then a PR with clear communication about the question is a good way to go.